sqlmesh/core/config/format.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlmesh.core.config.base import BaseConfig
+
+
+class UIFormatterConfig(BaseConfig):
+    """The UI formatter configuration for SQLMesh.
+
+    Args:
+        enabled: Whether to enable the UI auto formatter or not.
+    """
+
+    enabled: bool = True
+
+
+class FormatConfig(BaseConfig):
+    """The format configuration for SQL code.
+
+    Args:
+        normalize: Whether to normalize the SQL code or not.
+        pad: The number of spaces to use for padding.
+        indent: The number of spaces to use for indentation.
+        normalize_functions: Whether or not to normalize all function names. Possible values are: 'upper', 'lower'
+        leading_comma: Whether to use leading commas or not.
+        max_text_width: The maximum text width in a segment before creating new lines.
+        append_newline: Whether to append a newline to the end of the file or not.
+        ui: The UI formatter configuration.
+    """
+
+    normalize: bool = False
+    pad: int = 2
+    indent: int = 2
+    normalize_functions: t.Optional[str] = None
+    leading_comma: bool = False
+    max_text_width: int = 80
+    append_newline: bool = False
+    ui: UIFormatterConfig = UIFormatterConfig()
+
+    @property
+    def generator_options(self) -> t.Dict[str, t.Any]:
+        """Options which can be passed through to the SQLGlot Generator class.
+
+        Returns:
+            The generator options.
+        """
+        return {
```

Comment:
Can't this be just `self.dict(exclude={"append_newline"})` just like you did before?

Reply:
Ya it's just something you have to remember to do if trying to use the API. Instead of a prop which does it for you in 1 place that's easy to change or extend. 

Whichever way you prefer I can keep.

Reply:
No, `generator_options` is fine, I like it. My question is mostly why not use `self.dict(...)` in its body?

Reply:
Oh I see, yes thats fair